test_roscpp: added unit test for calling zombie services, see #434

xqms opened the pull request:
Zombie services are left behind by crashed nodes. Currently roscpp can
lock up when you call such a zombie service. This adds a unit test (which currently fails).

See #434 for details.

xqms committed:
test_roscpp: added unit test for calling zombie services, see #434

Zombie services are left behind by crashed nodes. Currently roscpp can
lock up when you call such a zombie service.

ros-pull-request-builder commented:
Can one of the admins verify this patch?

dirk-thomas commented:
@ros-pull-request-builder

ros-pull-request-builder commented:
Test failed.
Refer to this link for build results: http://jenkins.ros.org/job/_pull_request-indigo-ros_comm/22/

dirk-thomas commented:
The CI fails for the new test which is intentional. I will wait to merge this until the PR also includes a fix for it.

But I don't expect to have time to work on this soon. So if someone would like to look into this that would be very welcome.

xqms committed:
test_roscpp: service_call_zombie: reduce service call count

This makes the test faster if it passes.

xqms committed:
roscpp: Connection: emit drop_signal_ after unlocking drop_mutex_

Callbacks connected to the signal might want to lock other things, which
will violate lock ordering and probably cause a deadlock.

Moving the call out of the protected section is not dangerous, since
the connection is only dropped once.

xqms committed:
roscpp: ServiceManager: initialize connection before connecting, fixes #434

This makes sure that the drop callback is registered before the connect
occurs. Previously it was possible to loose the drop event and cause a
freeze of the current thread.

ros-pull-request-builder commented:
Test passed.
Refer to this link for build results: http://jenkins.ros.org/job/_pull_request-indigo-ros_comm/27/

dirk-thomas commented:
Thank you for the patch as well as the test!

xqms commented:
Thanks for merging. Do you have an estimate on when this will hit the repositories?

dirk-thomas commented:
`ros_comm` has been released into _Indigo_ due to a different patch necessary. So it should be in the shadow-fixed repo within one day. Public depends fully on if there are regressions in any packages - if not it might be a couple of days.
